### PR TITLE
Polish metrics package

### PR DESCRIPTION
**Pull request by sp-jordi-forns**
- [x] Depends on #34

### CODE REVIEW

File: metrics/example_test.go [hunk 1]
```diff
@@ -43,9 +42,20 @@ func ExampleEvent() {
...
-	// and UDP server and then close it it will fail as expected.
+func ExampleTimer() {
+	discardAllMetrics := metrics.NewDiscardAll()
+	go discardAllMetrics.Run(context.Background())
+
+	timer := discardAllMetrics.Timer("test.timer")
+	timer.Start()
+	timer.Stop()
+
+	// Output:
```
Comment (gnieto) [inline metrics/example_test.go]: What this means?

File: metrics/example_test.go [hunk 2]
```diff
@@ -75,9 +85,9 @@ func ExampleStatsDBackend() {
 }
 
 func ExampleErrorHandler() {
-	errors := make(chan error)
+	errs := make(chan error)
```
Comment (gnieto) [inline metrics/example_test.go]: I don't like this, but, go ahead if there's consensus

File: metrics/example_test.go (hunk 1 above)
Comment (sp-jordi-forns) [reply metrics/example_test.go]: it will complain if not present

File: metrics/gostats_internal_test.go [hunk 3]
```diff
@@ -1,58 +1,4 @@
...
-
-func TestGoStats(t *testing.T) {
-	assert := assert.New(t)
-	numGoMetrics := reflect.TypeOf(goMetrics{}).NumField()
-	rec := make(recorder, numGoMetrics)
-
-	duration := time.Millisecond * 100
-	publisher := NewPublisher(rec, StatsDEncoder, duration*2, nil)
-	ctx := context.Background()
-	go publisher.Run(ctx)
```
Comment (sp-daniel-pinyol) [inline metrics/gostats_internal_test.go]: don't we need to stop the publisher & runner runners? may with context timeout?

File: metrics/gostats_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+func TestGoStats(t *testing.T) {
+	assert := assert.New(t)
+	numGoMetrics := reflect.TypeOf(metrics.GoMetrics{}).NumField()
+	rec := make(recorder, numGoMetrics)
+
+	duration := time.Millisecond * 100
+	publisher := metrics.NewPublisher(rec, metrics.StatsDEncoder, duration*2, nil)
+	ctx := context.Background()
+	go publisher.Run(ctx)
```
Comment (sp-daniel-pinyol) [inline metrics/gostats_test.go]: don't we need to stop the publisher & runner runners? may with context timeout?

**Comment by sp-jordi-forns** [comment]
@sp-daniel-pinyol added cancellation to the context in the test 60aac79d256dfbe37551a63eeba2a020af704038

**Pull request merged**